src/topicButton.js

```diff
@@ -18,8 +22,14 @@ var TopicButton = GObject.registerClass({
 }, class TopicButton extends Gtk.Frame {
     _init(props = {}) {
         super._init(props);
-
         const overlay = new Gtk.Overlay();
+
+        if (this._lockscreen) {
+            this.topicLock = new Lockscreen({visible: true});
+            overlay.add_overlay(this.topicLock);
```

Comment:
I was surprised that the lockscreen widgets work when added as an overlay, instead of having the thing they are locking added as their child widget! But I think this is indeed the best way to do it. Otherwise you have to deal with widgets having different parents depending on whether there is a lockscreen or not.